src/Raven.Client/Documents/Commands/MultiGet/MultiGetCommand.cs

```diff
@@ -329,8 +327,24 @@ public void Dispose()
 
         private void DisposeCache()
         {
-            _cached?.Dispose();
-            _cached = null;
+            //If _cached is not null - it means that the client approached with this multitask request to node and the request failed.
+            //and now client tries to send it to another node.
+            if (_cached != null)
+            {
+                _cached.Dispose();
+                _cached = null;
+
+                // The client sends the commands.
+                // Some of which could be saved in cache with a response 
+                // that includes the change vector that received from the old fallen node.
+                // The client can't use those responses because their URLs are different 
+                // (include the IP and port of the old node), because of that the client 
+                // needs to get those docs again from the new node.
+                foreach (var command in _commands)
```

Comment:
Don't we need to check the cache and renew the value? Otherwise, we may _not_ use the cache after a failure, no?

Reply:
Done